### PR TITLE
refactor: Introduce table header th-element

### PR DESCRIPTION
**Pull request by pan-kot**
### Description

Similar to body-cell/td-element, the header-cell/th-element encapsulates cell styles that can be reused between regular and selection cells.

Depends on https://github.com/cloudscape-design/components/pull/1009

### How has this been tested?

Existing tests.

The VR regression is expected. It caused by moving isVisualRefresh check to the cell element itself when previously it came from the table. Now the style applies when the cell is rendered as standalone for permutations.

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by rubencarvalho — DISMISSED** [review]

**Review by rubencarvalho — DISMISSED** [review]

File: src/table/header-cell/th-element.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  sortable?: boolean;
+  sorted?: boolean;
+  sortingDescending?: boolean;
+  sortingDisabled?: boolean;
+  resizableColumns?: boolean;
+  hidden?: boolean;
+  sticky?: boolean;
+  stuck?: boolean;
+  stripedRows?: boolean;
+  hasSelection?: boolean;
```
Comment (just-boris) [inline src/table/header-cell/th-element.tsx:22]: Are these all optional?

Optional props lead to unused code, as it was demonstrated in the past: https://github.com/cloudscape-design/components/commit/368b042255e4fc257a2dab72b25c38818a16663c

If `undefined` is supported value, pass it as `| undefined`

File: src/table/header-cell/th-element.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import clsx from 'clsx';
+import React from 'react';
+import { TableProps } from '../interfaces';
+import { getAriaSort, getSortingStatus } from './utils';
+import styles from '../body-cell/styles.css.js';
+import rootStyles from '../styles.css.js';
```
Comment (just-boris) [inline src/table/header-cell/th-element.tsx:8]: If you want to extract a reusable component you need to isolate its styles too

File: src/table/header-cell/th-element.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import clsx from 'clsx';
+import React from 'react';
+import { TableProps } from '../interfaces';
+import { getAriaSort, getSortingStatus } from './utils';
+import styles from '../body-cell/styles.css.js';
```
Comment (just-boris) [inline src/table/header-cell/th-element.tsx:7]: `body-cell/styles.scss` file containing `.header-cell-*` classes is even more confusing than the previous revision.

If there are difficulties wrapping this refactoring, let's talk in DM

**Comment by pan-kot** [comment]
Closing in favour of https://github.com/cloudscape-design/components/pull/1022

**Pull request closed without merging**